Reneiski_Homework1/ReneiskiSelenium11/Program.cs

```diff
@@ -0,0 +1,246 @@
+
+            var userAge = "41";
+            var userAgeTextBox = _driver.FindElement(By.XPath("//*[@id='age']"));
+            userAgeTextBox.SendKeys(userAge);
+
+            var userSalary = "20000";
+            var userSalaryTextBox = _driver.FindElement(By.XPath("//*[@id='salary']"));
+            userSalaryTextBox.SendKeys(userSalary);
+
+            var userDepartment = "QA";
+            var userDepartmentTextBox = _driver.FindElement(By.XPath("//*[@id='department']"));
+            userDepartmentTextBox.SendKeys(userDepartment);
+
+            var submitButton = _driver.FindElement(By.XPath("//*[@id='submit']"));
+            submitButton.Click();
+
+            var currentResult1 = _driver.FindElement(By.XPath("//*[@class='rt-table']"));
+            var currentResultText1 = currentResult1.Text;
+
+            Assert.IsTrue(currentResultText1.Contains(firstName));
+            Assert.IsTrue(currentResultText1.Contains(lastName));
+            Assert.IsTrue(currentResultText1.Contains(userAge));
+            Assert.IsTrue(currentResultText1.Contains(userEmail));
+            Assert.IsTrue(currentResultText1.Contains(userSalary));
+            Assert.IsTrue(currentResultText1.Contains(userDepartment));
+
+            var editButton = _driver.FindElement(By.XPath("//*[@id='edit-record-4']"));
+            editButton.Click();
+
+            var editedUserSalary = "22000";
+            var editedUserSalaryTextBox = _driver.FindElement(By.XPath("//*[@id='salary']"));
+            editedUserSalaryTextBox.Clear();
+            editedUserSalaryTextBox.SendKeys(editedUserSalary);
+
+            var submitButtonAfterEdit = _driver.FindElement(By.XPath("//*[@id='submit']"));
+            submitButtonAfterEdit.Click();
+
+            var deleteButton = _driver.FindElement(By.XPath("//*[@id='delete-record-4']"));
+            deleteButton.Click();
+
+            var currentResult2 = _driver.FindElement(By.XPath("//*[@class='rt-table']"));
+            var currentResultText2 = currentResult2.Text;
+
+            Assert.IsFalse(currentResultText2.Contains(firstName));
+            Assert.IsFalse(currentResultText2.Contains(lastName));
+            Assert.IsFalse(currentResultText2.Contains(userAge));
+            Assert.IsFalse(currentResultText2.Contains(userEmail));
+            Assert.IsFalse(currentResultText2.Contains(userSalary));
+            Assert.IsFalse(currentResultText2.Contains(userDepartment));
+        }
+
+        [Test]
+        public void ForthButtonsTest()
+        {
+            _driver.Navigate().GoToUrl("https://demoqa.com/buttons");
+
+            var doubleClickButton = _driver.FindElement(By.XPath("//*[@id='doubleClickBtn']"));
+            _driverActions.DoubleClick(doubleClickButton).Perform();
+
+            var doubleClickResult = _driverWait.Until(drv => drv.FindElement(By.XPath("//*[@id='doubleClickMessage']")));
+            var doubleClickResultText = doubleClickResult.Text;
```

Comment:
this doubleClickResultText is expected to contain only "You have done a double click" text, so it's better to use Assert.AreEqual.
You need to use Assert.IsTrue(sometext.Contains(text)) only in cases when you expect that sometext will contain not only your text, but also some another text.
If your result has text like "You have done a double click and smth else lala", your test will pass, but there will be a bug because the expected result is exactly text "You have done a double click"
Please correct here and for all cases above/below

Reply:
Fixed